doc/man3/OSSL_PARAM_TYPE.pod

```diff
@@ -294,12 +411,23 @@ could fill in the parameters like this:
     const OSSL_PARAM *p;
 
     if ((p = OSSL_PARAM_locate(params, "foo")) == NULL)
-        OSSL_PARAM_set_utf8_ptr(p, "foo value");
+        error();
+    OSSL_PARAM_set_utf8_ptr(p, "foo value");
     if ((p = OSSL_PARAM_locate(params, "bar")) == NULL)
-        OSSL_PARAM_set_utf8_ptr(p, "bar value");
-    if ((p = OSSL_PARAM_locate(params, "cookie")) == NULL)
+        error();
+    OSSL_PARAM_set_utf8_ptr(p, "bar value");
+    if ((p = OSSL_PARAM_locate(params, "cookie")) != NULL)
         OSSL_PARAM_set_utf8_ptr(p, "cookie value");
 
+or, more succinctly, like this:
+
+    /* const OSSL_PARAM *params */
+
+    if (!OSSL_PARAM_locate_set_utf8_ptr(params, "foo", "foo value")
+            || !OSSL_PARAM_locate_set_utf8_string(params, "bar", "bar value"))
+        error();
+    OSSL_PARAM_optional_set_utf8_string(params, "cookie", "cookie value");
```

Comment:
``` C
    OSSL_PARAM_optional_set_utf8_ptr(params, "foo", "foo value");
    OSSL_PARAM_optional_set_utf8_string(params, "bar", "bar value");
    OSSL_PARAM_optional_set_utf8_string(params, "cookie", "cookie value");
```

Reply:
If you really want to make an valuable example, it would be to use a mixture of `locate_get` and `optional_get` functions.  Personally, I rather imagine that some parameters might have to be grouped together.  For example, that if "foo" is set by the caller, so must "bar".